openai-whisper: 20230124 -> 20230314

MayNiklas opened the pull request:
###### Description of changes

This week, two PR's got merged making this update possible:
We now have Tiktoken & Triton in NixPkgs!

This is a simple version bump, needing additional dependencies.

OpenAI introduced additional tests - some of them we can't run:
- they require network access
- they require access to the NVIDIA driver

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

MayNiklas commented:
I needed to make some small changes (I tested my previous PR with unfree packages being allowed).
For CUDA support, whisper now uses triton (same with torch).
That being said: it's an optional dependency, only needed on X86 systems with NVIDIA CUDA GPU's.
By adding a `cudaSupport` option, we only add triton when cudaSupport is enabled -> package still can be build without unfree packages when needed!

MayNiklas commented:
@mweinelt if you are okay with it, I would also like to add myself as a package maintainer for `openai-whisper`.
I'm working on a accessibility project relying on whisper -> I'm already very up to date with whisper and would love to be informed about new PR's relating whisper.

mweinelt commented on pkgs/development/python-modules/openai-whisper/default.nix:
```diff
@@ -39,13 +45,30 @@ buildPythonPackage rec {
   ];
 
+  postPatch = ''
+    sed -i 's/tiktoken==0.3.1/tiktoken>=0.3.1/' requirements.txt
```
Ideally using `substituteInPlace`, so we get a log entry when it stops applying.

MayNiklas replied:
Done!👍

> Ideally using `substituteInPlace`, so we get a log entry when it stops applying.

mweinelt reviewed (commented):
Yeah, feel free to maintain this package.

mweinelt commented:
The essential stuff should go into the commit message, because it gets distributed and is visible while browsing the logs or bisecting.

That includes the release notes, any motivation and whatever else you have to say. Also mentioned in [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes).

mweinelt commented on pkgs/development/python-modules/openai-whisper/default.nix:
```diff
@@ -56,7 +80,11 @@ buildPythonPackage rec {
+    "test_median_filter_equivalence"
   ];
 
   meta = with lib; {
```
```suggestion
  meta = with lib; {
    changelog = "https://github.com/openai/whisper/blob/v$[version}/CHANGELOG.md";
```

MayNiklas committed:
openai-whisper: 20230124 -> 20230314

- add cudaSupport option
- add required dependencies: numba, scipy, tiktoken
- add cuda related dependencies only when cudaSupport = true
https://github.com/openai/whisper/blob/v20230314/CHANGELOG.md

MayNiklas commented:
> The essential stuff should go into the commit message, because it gets distributed and is visible while browsing the logs or bisecting.
> 
> That includes the release notes, any motivation and whatever else you have to say. Also mentioned in [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes).

You're right! I completely forgot about this!
I changed the commit message!

mweinelt commented:
You forgot adding yourself as maintainer.

MayNiklas committed:
openai-whisper: add MayNiklas to maintainers
